compiler/pipes/calc-bad-vars.cpp

```diff
@@ -115,6 +115,343 @@ struct FuncCallGraph {
   }
 };
 
+using CallstackColors = std::vector<function_palette::color_t>;
+
+class Callstack {
+  using raw = std::vector<FunctionPtr>;
+  using raw_set = std::unordered_set<FunctionPtr>;
+  using it = std::vector<FunctionPtr>::iterator;
+  using element = FunctionPtr;
```

Comment:
As for me, such usings create more mess than benefit. It's really hard to read "it begin;" and others: you should always keep in mind, that "it" means.
I'd suggest not to use them at all.